modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCDataStoreTest.java

```diff
@@ -221,4 +225,27 @@ public JDBCReaderCallback createReaderCallback() {
         verify(callback, times(1)).afterNext(rowData, false);
         verify(callback, times(1)).finish(reader);
     }
+
+    @Test
+    public void testGetConnectionAutocommit() throws Exception {
+
+        JDBCMockObjectFactory jdbcMock = new JDBCMockObjectFactory();
+        MockDataSource dataSource = jdbcMock.getMockDataSource();
+        dataSource.setupConnection(jdbcMock.getMockConnection());
+        JDBCDataStore store = new JDBCDataStore();
+        store.setSQLDialect(mock(BasicSQLDialect.class));
+        store.setDataSource(dataSource);
+
+        try (Connection conn = store.getConnection(Transaction.AUTO_COMMIT)) {
+            Assert.assertEquals(Boolean.TRUE, conn.getAutoCommit());
+        }
+
+        try (Transaction transaction = new DefaultTransaction();
+                Connection conn2 = store.getConnection(transaction)) {
+            Assert.assertEquals(Boolean.FALSE, conn2.getAutoCommit());
+        }
+
+        Transaction transaction = null;
```

Comment:
PMD is a little dense here:

```
2023-01-14T15:44:59.3686060Z [INFO] PMD Failure: org.geotools.jdbc.JDBCDataStoreTest:248 Rule:CloseResource Priority:3 Ensure that resources like this Transaction object are closed after use.
```

Probably easier to just pass "null" to the getConnection method without holding a variable...

Reply:
Passing null will not compile, because there is two getConnection method's. It would be getConnection((Transaction)null).
The PMD is complaining about a missing close, so I'll fix it with a try-with-resources. Maybe that PMD would allow a SuppressWarnings attribute as well.

Reply:
Yes, PMD allows for a SuppressWarning indeed, it's explained in the QA page, linked also from the checklist above:
https://docs.geotools.org/latest/developer/conventions/code/qa.html